x-pack/test/security_solution_cypress/cypress/e2e/investigations/timelines/unified_components/query_tab.cy.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { CELL_ADD_TO_TIMELINE_BUTTON } from '../../../../screens/alerts';
+import {
+  openEventDetailsFlyout,
+  openHostDetailsFlyout,
+  openUserDetailsFlyout,
+} from '../../../../tasks/unified_timeline';
+import {
+  CELL_FILTER_IN_BUTTON,
+  GET_UNIFIED_DATA_GRID_CELL,
+  GET_UNIFIED_DATA_GRID_CELL_HEADER,
+  TIMELINE_DETAILS_FLYOUT,
+  TIMELINE_DETAILS_FLYOUT_CLOSE_BTN,
+} from '../../../../screens/unified_timeline';
+import {
+  TIMELINE_DATA_PROVIDERS_CONTAINER,
+  TIMELINE_FILTER_BADGE,
+} from '../../../../screens/timeline';
+import { GET_DISCOVER_DATA_GRID_CELL_HEADER } from '../../../../screens/discover';
+import { addFieldToTable, removeFieldFromTable } from '../../../../tasks/discover';
+import { login } from '../../../../tasks/login';
+import { visitWithTimeRange } from '../../../../tasks/navigation';
+import { openTimelineUsingToggle } from '../../../../tasks/security_main';
+import { createNewTimeline, executeTimelineSearch } from '../../../../tasks/timeline';
+import { ALERTS_URL } from '../../../../urls/navigation';
+
+describe(
+  'Unsaved Timeline query tab',
+  {
+    tags: ['@ess', '@serverless'],
+    env: {
+      ftrConfig: {
+        kbnServerArgs: [
+          `--xpack.securitySolution.enableExperimental=${JSON.stringify([
+            'unifiedComponentsInTimelineEnabled',
+          ])}`,
+        ],
+      },
+    },
+  },
+  () => {
+    beforeEach(() => {
+      login();
+      visitWithTimeRange(ALERTS_URL);
+      openTimelineUsingToggle();
+      createNewTimeline();
+      executeTimelineSearch('*');
+    });
+    it('should be able to add/remove columns correctly', () => {
+      cy.get(GET_UNIFIED_DATA_GRID_CELL_HEADER('agent.type')).should('not.exist');
+      addFieldToTable('agent.type');
+      cy.get(GET_DISCOVER_DATA_GRID_CELL_HEADER('agent.type')).should('be.visible');
+      removeFieldFromTable('agent.type');
+      cy.get(GET_DISCOVER_DATA_GRID_CELL_HEADER('agent.type')).should('not.exist');
+    });
+
+    context('hover actions', () => {
+      it('should add to timeline successfully', () => {
+        cy.get(GET_UNIFIED_DATA_GRID_CELL('host.name', 0)).then(($el) => {
+          const hostName = $el.text();
+          cy.get(GET_UNIFIED_DATA_GRID_CELL('host.name', 0)).realHover();
```

Comment:
Please wrap the actions in tasks.

Lately we found `realHover` might be a cause of flakiness, try to use `.trigger('mouseover')` instead.

Reply:
Actually, in this case `realHover` seems to be working better and `mouseover` is not working at all.. Still looking at it why.

Reply:
So... mouseover and realhover both seems flaky so I removed these tests for now so that this PR can go through and i will add those as either unit/cypress tests in follow up PR.

But I have confirmed that other tests are not flaky.